scripts/kfctl.sh

```diff
@@ -260,6 +264,9 @@ parseArgs() {
       --skipInitProject)
         SKIP_INIT_PROJECT=true
         ;;
+      --skipCreatePipelinePersistentStorage)
```

Comment:
Why the flag? Can we follow the pattern of
kfctl generate  
Performs the default desired behavior

For non-default behavior; customize the configs after running kfctl generate?

Reply:
do we expect user to change env.sh manually? If so I don't have to pass this flag here. 

I'm not sure what can be added to the args here what can't. it looks like there are a few args here are also in env.sh. If we expect user to change env.sh directly, when will they pass them directly?

Reply:
The user can customize the behavior via the YAML file storage-kubeflow.yaml right?
e.g. via the option createPipelinePersistentStorage and others?

So the pattern we'd like to follow is kfctl takes no arguments related to this.
kfctl.sh generate creates the storage-kubeflow.yaml file with suitable defaults.
User can then modify that YAML file as needed to get desired behavior; e.g. use PD or not.

Reply:
the flag in storage-kubeflow.yaml is specific for DM, while whether creating a persistent storage  would impact what parameter to pass to ks app downstream. The flag should be declared and used at a higher level. and that's why i put it in env variable. Is there any better location for that flag?

Reply:
Why do you need a higher level flag? 

Could you just tell users to do

ks init
ks generate
1. edit storage-kubeflow.yaml
2. Run  
    ks param set pipeline mysqlPd "${DEPLOYMENT_NAME}-storage-pipeline-db"
    ks param set pipeline nfsPd "${DEPLOYMENT_NAME}-storage-pipeline-nfs"

The problem with introducing a new command line flag is what's the source of truth?
Is the environment variable which is persisted in env.sh or is it the DM and ksonnet configs?

My preference would be to validate that customers are actually use it and then figure out how to add sugar to make it simpler.

Reply:
done.